### PR TITLE
fix(aws-ecs): check for invalid capacityProviderName

### LINKED ISSUE #17321: (aws-ecs): in case you define the name of ecs capacity provider start with `aws`, `ecs` or `fargate` will create fail.
### Description

in case you define the name of ecs capacity provider start with `aws`, `ecs` or `fargate` will create fail.

### Use Case

if user define the name of ecs capacity provider, will pre-check before deploy.

### Proposed Solution

add pre-check function.

### Other information

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-capacityprovider.html

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by neilkuan**
fix(aws-ecs): add precheck name of capacity provider when specify it.
chore(aws-ecs): update capacityProviderName Description

close #17321
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by neilkuan** [comment]
Hi @peterwoodworth, do you have a free to take a look this PR, thank you. :)

**Comment by peterwoodworth** [comment]
Thanks for the contribution @neilkuan! is there any original issue which this is meant to close? And if not, could you create one to describe the issue that needs to be fixed? Thanks

**Comment by neilkuan** [comment]
thank @peterwoodworth , already create the issue #17321.

**Comment by peterwoodworth** [comment]
I think this looks good - I don't think this would be a breaking change. @madeline-k should verify though

**Comment by neilkuan** [comment]
Hi @madeline-k , do you have a free to take a look this PR, thank you. :)

**Comment by neilkuan** [comment]
Hi @peterwoodworth, actually this PR adds a pre-check in CDK to avoid possible deployment failure and does not add any new features. It doesn't make sense to update the README to pass the lint. Let's classify it as chore or fix. wdyt?

File: packages/@aws-cdk/aws-ecs/test/cluster.test.ts [hunk 1]
```diff
@@ -2245,4 +2245,43 @@ test('can add ASG capacity via Capacity Provider by not specifying machineImageT
     DefaultCapacityProviderStrategy: [],
   });
 
+});
+
+test('can get error when ASG Capacity Provider define capacityProviderName start with aws, ecs or faragte', () => {
```
Comment (pahud) [inline packages/@aws-cdk/aws-ecs/test/cluster.test.ts:2250]:
```suggestion
test('throws when ASG Capacity Provider with capacityProviderName starting with aws, ecs or faragte', () => {
```

File: packages/@aws-cdk/aws-ecs/test/cluster.test.ts [hunk 2]
```diff
@@ -2245,4 +2245,43 @@ test('can add ASG capacity via Capacity Provider by not specifying machineImageT
...
+  // THEN
+  expect(() => {
+    // WHEN Capacity Provider define capacityProviderName start with aws.
+    const capacityProviderAl2 = new ecs.AsgCapacityProvider(stack, 'provideral2', {
+      autoScalingGroup: autoScalingGroupAl2,
+      enableManagedTerminationProtection: false,
+      capacityProviderName: 'awscp',
+    });
+
+    // Ensure not added twice
```
Comment (pahud) [inline packages/@aws-cdk/aws-ecs/test/cluster.test.ts:2272]:
```suggestion
```
Let's just remove this comment.

File: packages/@aws-cdk/aws-ecs/test/cluster.test.ts [hunk 3]
```diff
@@ -2245,4 +2245,43 @@ test('can add ASG capacity via Capacity Provider by not specifying machineImageT
...
+
+  expect(() => {
+    // WHEN Capacity Provider define capacityProviderName start with ecs.
+    const capacityProviderAl2 = new ecs.AsgCapacityProvider(stack, 'provideral2-2', {
+      autoScalingGroup: autoScalingGroupAl2,
+      enableManagedTerminationProtection: false,
+      capacityProviderName: 'ecscp',
+    });
+
+    // Ensure not added twice
```
Comment (pahud) [inline packages/@aws-cdk/aws-ecs/test/cluster.test.ts:2284]:
```suggestion
```
Let's just remove this comment.

**Review by pahud — CHANGES_REQUESTED** [review]

**Review by pahud — COMMENTED** [review]
LGTM

**Comment by pahud** [comment]
Hi @madeline-k can you take a look please?

**Comment by upparekh** [comment]
Hi @neilkuan , can you please resolve the conflicts before we can merge this PR?

**Review by upparekh — APPROVED** [review]

**Pull request merged**